### PR TITLE
[Teacher] Fix launching Speed Grader in side-by-side

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-13046
affects: teacher
release note: Fixed a bug that could cause the wrong submission to display in Speed Grader

Test plan:
* Since I mucked with test schemes and CoreUITests I'll kick off the nightly build- it should pass.
* See steps in [ticket](https://instructure.atlassian.net/browse/MBL-13046)

### CODE REVIEW

File: Core/CoreUITests/CoreUITestCase.swift [hunk 1]
```diff
@@ -303,6 +303,13 @@ open class CoreUITestCase: XCTestCase {
         mockURL(url, data: data, response: response, error: error, noCallback: noCallback)
     }
 
+    open func mockGraphQL(_ json: Any) {
+        let data = try! JSONSerialization.data(withJSONObject: json, options: [])
+        let api = URLSessionAPI()
+        let url = URL(string: "/api/graphql", relativeTo: api.baseURL)!
+        mockURL(url, data: data, response: nil, error: nil, noCallback: false)
+    }
```
Comment (cobbal) [inline Core/CoreUITests/CoreUITestCase.swift]: Some way to mock more than 1 graphQL query at a time would probably be good, but that's a future bridge to burn.

File: rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    func testSpeedGrader() {
+        XCUIDevice.shared.orientation = .landscapeLeft
+        let course = mock(course: APICourse.make())
+        let enrollment1 = APIEnrollment.make(id: "1", user_id: "1", user: .make(id: "1", name: "User 1", sortable_name: "User 1", short_name: "User 1"))
+        let enrollment2 = APIEnrollment.make(id: "2", user_id: "2", user: .make(id: "2", name: "User 2", sortable_name: "User 2", short_name: "User 2"))
+        let enrollment3 = APIEnrollment.make(id: "3", user_id: "3", user: .make(id: "3", name: "User 3", sortable_name: "User 3", short_name: "User 3"))
+        let submissions = [
+            APISubmission.make(id: "1", user_id: "1", user: .make(id: "1", name: "User 1", short_name: "User 1")),
+            APISubmission.make(id: "2", user_id: "2", user: .make(id: "2", name: "User 2", short_name: "User 1")),
```
Comment (cobbal) [inline rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift]:
```suggestion
            APISubmission.make(id: "2", user_id: "2", user: .make(id: "2", name: "User 2", short_name: "User 2")),
```

File: rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+            APISubmission.make(id: "3", user_id: "3", user: .make(id: "3", name: "User 3", short_name: "User 1")),
```
Comment (cobbal) [inline rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift]:
```suggestion
            APISubmission.make(id: "3", user_id: "3", user: .make(id: "3", name: "User 3", short_name: "User 3")),
```

**Comment by armstrongnate** [comment]
Nightly build:
* https://app.bitrise.io/build/f2e164acc6f462fe#?tab=log

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Fixed a bug that could cause the wrong submission to display in Speed Grader

#### Affected Apps: Teacher

[MBL-13046](https://instructure.atlassian.net/browse/MBL-13046)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift:27:5: Function Body Length Violation: Function body should span 100 lines or less excluding comments and whitespace: currently spans 104 lines (function_body_length)
    /Users/vagrant/git/rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift:171:30: warning: Trailing Comma Violation: Multi-line collection literals should have trailing commas. (trailing_comma)
    /Users/vagrant/git/rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift:192:58: warning: Trailing Comma Violation: Multi-line collection literals should have trailing commas. (trailing_comma)
    ⚠️  /Users/vagrant/git/rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift:206:58: Trailing Comma Violation: Multi-line collection literals should have trailing commas. (trailing_comma)
    /Users/vagrant/git/rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift:220:58: warning: Trailing Comma Violation: Multi-line collection literals should have trailing commas. (trailing_comma)
    /Users/vagrant/git/rn/Teacher/ios/TeacherUITests/SpeedGrader/IPadSpeedGraderTests.swift:227:10: warning: Trailing Comma Violation: Multi-line collection literals should have trailing commas. (trailing_comma)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by cobbal** [comment]
Looks great. Once the builds finish and the lint is fixed I'll approve.

**Pull request merged**